src/x/debug/triggering_profile.go

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package debug
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"runtime/pprof"
+	"text/template"
+	"time"
+
+	"github.com/m3db/m3/src/x/instrument"
+
+	"go.uber.org/zap"
+)
+
+const (
+	// ContinuousCPUProfileName is the name of continuous CPU profile.
+	ContinuousCPUProfileName = "cpu"
+)
+
+var (
+	defaultConditional     = func() bool { return true }
+	defaultInterval        = time.Second
+	errNoFilePathTemplate  = errors.New("no file path template")
+	errProfileName         = errors.New("no profile name")
```

Comment:
Nit: Call this errNoProfileName for consistency.

Reply:
Sg